templates/client-sdk-release/template.yaml

```
@@ -0,0 +1,106 @@
+
+  - name: 'sdkPackageName'
+    type: string
+    default: ''
+
+  - name: 'dependsOn'
+    type: object
+    default: []
+
+# ------------------
+# Test parameters only
+
+  - name: 'isTestEnvironment'
+    type: boolean
+    default: false
+
+  - name: 'npmRegistry'
+    type: string
+    default: ''
+
+  - name: 'artifactName'
+    type: string
+    default: Bundle_SDK
+
+# ------------------
+# Steps 
+
+steps:
+
+#Setting variables:
+# codegenPackageNameParameter : string to setup generated package name
+# generatedCodeDir: generated package directory
+# 
+# If test:
+# setNpmRegistryValue: string to setup npm registry
+- bash: |
+    # Set SDK package name, if needed
+    [ "${{parameters.sdkPackageName}}" == "" ] && PKG_NAME_PARAM='' || PKG_NAME_PARAM='--package-name ${{ parameters.sdkPackageName }}'     
+    echo "setting codegenPackageNameParameter=$PKG_NAME_PARAM" 
+    echo "##vso[task.setvariable variable=codegenPackageNameParameter;]$PKG_NAME_PARAM"
+    
+    #Set generated code base directory
+    echo "setting generatedCodeDir=$(Agent.TempDirectory)/generated" 
+    echo "##vso[task.setvariable variable=generatedCodeDir;]$(Agent.TempDirectory)/generated"
+
+    # Set NPM Registry, if test
+    [ "${{ parameters.isTestEnvironment }}" == "True" ] && NPM_REGISTRY='--registry ${{ parameters.npmRegistry }}' || NPM_REGISTRY=''     
+    echo "setting setNpmRegistryValue=$NPM_REGISTRY" 
+    echo "--> ${{ parameters.isTestEnvironment }}" 
+    echo "##vso[task.setvariable variable=setNpmRegistryValue;]$NPM_REGISTRY"
+  name: setvarStep
+  displayName: 'Setup conditional variables'
+- bash: |
+    npx -p @pagopa/openapi-codegen-ts gen-api-sdk --api-spec ${{ parameters.openapiSpecPath }} --out-dir $(generatedCodeDir) $(codegenPackageNameParameter) --client
+  displayName: 'Generate client and definitions'
+  workingDirectory: ${{ parameters.projectDir }}
+
+- script: |
+      yarn install
+  displayName: 'Install depencencies'
+  workingDirectory: $(generatedCodeDir)
+
+- script: |
+      yarn build
+  displayName: 'Build depencencies'
+  workingDirectory: $(generatedCodeDir)
+
+# - task: CopyFiles@2
+#   inputs:
+#     SourceFolder: '$(generatedCodeDir)'
+#     TargetFolder: '$(System.DefaultWorkingDirectory)/bundle_sdk'
+#     # The list of files to be added to the bundle
+#     # We need to add .ts and tsconfig.json files because of the 'prepublish' script within
+#     # the generate package.json file (as visible in the template here:  https://github.com/pagopa/io-utils/blob/master/templates/sdk/package.json.njk#L10 )
+#     Contents: |
+#       ?(*.ts|*.js|*.js.map)
+#       package.json
+#       tsconfig.json
+#       openapi.yaml
+#     flattenFolders: true
+#   displayName: 'Copy bundle files'
+
+#Publish artifact so we can check it
+- publish: $(generatedCodeDir)
+  artifact: ${{ parameters.artifactName }}
+
+
+- task: Npm@1
+  inputs: 
+    command: custom
+    customCommand: publish --access public $(setNpmRegistryValue)
```

Apply this if we decide to use the npm registry as default for npmRegistry
```suggestion
    customCommand: publish --access public --registry $(npmRegistry)
```